viewing_party/party.py

```diff
@@ -1,23 +1,198 @@
+
+
+def get_most_watched_genre(user_data):
+    """
+    Return the most watched genre in the users watched list.
+        Args:
+            user_data (dict): A dict of movies the user has watched.
+        Returns:
+            most_watched_genre (string): The users most watched genre.
+    """
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        genre_count = {}
+        for movie in user_data["watched"]:
+            if movie["genre"] in genre_count:
+                genre_count[movie["genre"]] += 1
+            else:
+                genre_count[movie["genre"]] = 1
+
+        most_watched_genre = max(genre_count, key=genre_count.get)
+        return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    """
+    Return the movies the user has watched but none of their friends have watched.
+        Args:
+            user_data (dict): A dict of movies the user has watched and 
+            their friends have watched. 
+        Returns:
+            user_unique_watched (list): Movies only the user has watched.
+    """
+    user_unique_watched = user_data["watched"].copy()
+
+    for friend in user_data["friends"]:
+        for user_movie in user_data["watched"]:
+            if user_movie in friend["watched"] and (
+            user_movie in user_unique_watched):
+                user_unique_watched.remove(user_movie)
+
+    return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    """
+    Return the movies at least one of the user's friends have watched but the
+    user has not watched.
+        Args:
+            user_data (dict): A dict of movies the user has watched and 
+            their friends have watched. 
+        Returns:
+            friend_unique_watched (list): Movies only the users friends 
+            have watched.
+    """
+    friends_unique_watched = []
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie not in user_data["watched"] and (
+            friend_movie not in friends_unique_watched):
+                friends_unique_watched.append(friend_movie)
+    
+    return friends_unique_watched
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    """
+    Return a list of recommended movies for the user based on their subscriptions.
+        Args:
+            user_data (dict): A dict of the users watched movies and subscriptions,
+            and their friends' watched movies.
+        Returns:
+            recommended_movies (list): Movies only the user friends have watched
+            and that are available in the users subscriptions.
+    """
+    recommended_movies = []
+
+    for friend in user_data["friends"]:

```

Comment:
👍 

But we also have a function that we wrote that gets the movies that only the friends have watched which could simplify our logic here!